src/main/java/net/snowflake/client/jdbc/telemetry/TelemetryClient.java

```diff
@@ -329,8 +331,8 @@ private boolean sendBatch() throws IOException {
                 ? HttpUtil.executeGeneralRequest(
                     post,
                     TELEMETRY_HTTP_RETRY_TIMEOUT_IN_SEC,
-                    this.session.getAuthTimeout(),
```

Comment:
Could you keep the current session based logic when a session exists and use your new defaults when `this.session` is null?

Reply:
I'm a little confused, I think that's what I did (we check whether `this.session== null` first), could you elaborate?

Reply:
Maybe I'm missing something, but I see a null check [here](https://github.com/snowflakedb/snowflake-jdbc/pull/918/files#diff-5fc45b6960e1d85594aa410a363e3d2f87805f077af5f79a34595982160b7a1cR316), but this line where we're going from using `this.session.getAuthTimeout()` and `this.session.getHttpClientSocketTimeout()`  to hard coded constants is not in that if/else block

Reply:
yeah, there is another null check at line 328 :)